server/src/internalClusterTest/java/org/elasticsearch/indices/IndicesLifecycleListenerIT.java

```diff
@@ -143,6 +144,21 @@ public void beforeIndexCreated(Index index, Settings indexSettings) {
         }
         updateIndexSettings(Settings.builder().put(INDEX_ROUTING_EXCLUDE_GROUP_PREFIX + "._name", node1), "index1");
         ensureGreen("index1");
+
+        var state = clusterAdmin().prepareState().get().getState();
+        logger.info("Final routing is {}", state.getRoutingNodes().toString());
+        var shard = findShard(state, "index1");
+        assertThat(shard, notNullValue());
+        assertThat(shard.state(), equalTo(ShardRoutingState.STARTED));
+        assertThat(findNodeById(state, shard.currentNodeId()).getName(), equalTo(node1));
+    }
+
+    private static ShardRouting findShard(ClusterState state, String name) {
```

Comment:
nit. I am not sure I like these short functions personally. I would prefer to see the code inside the test, since it's more straightforward to read the test that way.